tests/e2e/snapshot_stretched_supervisor.go

```diff
@@ -0,0 +1,1101 @@
+	   a) ZonalPolicy-immediateBindingMode
+	   b) ZonalPolicy-lateBinding (WFFC)
+	   2. Create a Statefulset with node affinity rule set to any particular zone (zone-1) with replica
+	   count set to 3 using the storage class (ZonalPolicy-lateBinding).
+	   3. Wait for PVC to reach the Bound state and Pods to reach the Running state.
+	   4. Write some data to volume
+	   5. Describe PV and verify the node affinity details should show up as  topology.kubernetes.io/zone in [zone-1]
+	   6. Make sure Pod is scheduled on appropriate nodes preset in the availability zone
+	   7. Get VolumeSnapshotClass "volumesnapshotclass-delete" from supervisor cluster.
+	   8. Create 3 volume snapshots (vols-1, vols-2, vols-3) for all 3 statefulset PVCs created above.
+	   9. Snapshot Verification: Execute and verify the steps mentioned in the Create snapshot mandatory checks
+	   10. Create 3 new PVCs using the volume snapshot created in step #8
+	   11. Wait for all new PVCs to reach the Bound state.
+	   12. Describe PV and verify the node affinity details should show up as Ex: topology.kubernetes.io/zone in [zone-1]
+	   13. Cleanup: Execute and verify the steps mentioned in the Delete snapshot mandatory checks
+	*/
+
+	ginkgo.It("Taking snapshot of statefulset pvcs and attaching "+
+		"it to a deployment pod", ginkgo.Label(p0, vanilla, block, wcp, core), func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		scParameters[svStorageClassName] = zonalPolicy
+
+		ginkgo.By("Create storage class")
+		storageclass, err := createStorageClass(client, scParameters, nil, "", "", false, zonalPolicy)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating StorageClass for Statefulset")
+		stsReplicas := 3
+		depReplica := 1
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+
+		framework.Logf("Create StatefulSet")
+		statefulset := createCustomisedStatefulSets(ctx, client, namespace, true,
+			int32(stsReplicas), true, allowedTopologies, true, true,
+			zonalPolicy, "", storageclass, zonalPolicy)
+		defer func() {
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
+		}()
+
+		framework.Logf("Verify PV node affinity and that the PODS are running on appropriate node")
+		err = verifyPVnodeAffinityAndPODnodedetailsForStatefulsetsLevel5(ctx, client, statefulset,
+			namespace, allowedTopologies, false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+			fmt.Sprintf("error verifying PV node affinity and POD node details: %v", err))
+
+		ssPodsBeforeScaleDown := fss.GetPodList(ctx, client, statefulset)
+
+		framework.Logf("Fetching pod 1, pvc1 and pv1 details")
+		pod1, err := client.CoreV1().Pods(namespace).Get(ctx,
+			ssPodsBeforeScaleDown.Items[0].Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pvc1 := pod1.Spec.Volumes[0].PersistentVolumeClaim
```

Comment:
Can we do this flow for 1 statefulset , that should be enough

Reply:
done